docs/papers/architecture-whitepaper.md

```diff
@@ -0,0 +1,1712 @@
+- **Add Inventory Item** - refers to adding a new inventory to the wallet. 
+- **Delete Inventory Item** - is the sub-capability of deleting an existing inventory.
+- **Update Inventory Item** - refers to the sub-capability of updating existing inventory.
+- **Refresh Inventory Item** - is a sub-capability specific for money and objects in the context of money (bank balances, cryptocurrencies) and objects (digital assets).
+
+###### Considerations
+- **Customized View** - As the size of the inventory grows, it might be required for the wallet owner to sort, categorize, and customize the inventory items, which would be a requirement on the API level to enable this functionality.
+- **Asynchronous query** - Since some of the inventory items require remote queries to external services (i.e., Banking, Brokerage), an asynchronous function to run the query could be considered.
+
+##### Connections
+**Connections** refer to the ability to establish and manage secure links that facilitate communication with other entities, such as individuals, organizations, services, and devices. It is the capability that enables secure transactions with other entities such as messaging, receiving credentials, or presenting proofs. 
+
+```mermaid
+sequenceDiagram
+  participant A as Agent A
+  participant B as Agent B
+  participant UB as User B
+  
+  A->>B: Request Connection to Agent B
+  B->>UB: Message: Agent A wants to connect with you
+  UB->>B: Consent connection request
+  B->>A: Response connection request
+  A->>A: Store connection info
+  A->>B: Send acknowledgement
+  B->>B: Store connection info
+  
+  A->>B: Transact
+  B->>A: response
+```
+###### Sub-capabilities
+- **Query Connections** - is a sub-capability for inquiring existing connections 
+- **Add Connection** - refers to the ability of adding a new connection to transact with the entity that the agent makes a connection with.
+- **Update Connection** - is the ability to update an existing connection. Updates can be related to, i.e., key rotation, change of service endpoints, or a new symmetric key negotiation.
+- **Delete Connection** - is a sub-capability to delete a connection from existing connections. 
+
+###### Considerations
+- **Role of Consent** - For human-centric wallets, [consent](#consents) is a required capability to ensure that the connection that is being established is according to the consent given by the wallet owner.
+- **Role of Policy** - Consent to establish a connection can be automated with [policy rules](#policy-engine). For machine-centric wallets, this is a required capability to ensure that connections that are being established are according to the pre-established policies.
+- **Trust and Reputation** - Mechanisms should be in place to establish and assess the trustworthiness of connections, possibly through a chain of trust or a web of trust. 
+
+##### Wallet Transfer and Recovery
```

Comment:
Not sure if it's relevant at this stage, but there's a case for considering future Decentralized Key Recovery Protocols.  At least in the web3 space new protocols based on Shamir Sharded Keys are popping up (i.e https://derecalliance.org/).

Reply:
Thanks for this comment. This is something that we have thought about for a future version that focuses on key recovery patterns. I will make an update to add this to the considerations section though.